### DISCUSSION TITLE
RFC: BFormFile closer to Bootstrap-vue?

### DISCUSSION THREAD

**Discussion by VividLemon** [cdmoro/bootstrap-vue-3#1213] [Polls]
Bootstrap-vue's implementation of BFormFile was due to Bootstrap v4 not having a "nice looking" appearance for file inputs. In Bootstrap v4, the appearance is what Bootstrap-vue's BFormFile looks like with their prop 'plain' activated. 

![image](https://github.com/bootstrap-vue-next/bootstrap-vue-next/assets/31498096/ddfbe7cf-12ae-46d3-a21e-e77557ea2c44)

Yuck!

Bootstrap v5 by default includes a better style for this. 

![image](https://github.com/bootstrap-vue-next/bootstrap-vue-next/assets/31498096/9feb9427-129f-411f-a4b8-e37ac7a50ca0)

I like this style, and it takes little to build and maintain, as the browser does most of the heavy lifting.

However, the current implementation of BFormFile on Bootstrap-vue-next lacks some of the features that bootstrap-vue has. Namely, visual feedback for drag/drop support (which can be added without any other major changes).

Particularly, the biggest issue is changing the "browse" button on the side, and the text on the inside. This is not bootstrap's responsibility, this is solely on the browser.

The way Bootstrap-vue, and many other css frameworks (like Vuetify), solve this by essentially creating an invisible input element, then show a element above that input element that can then modify the text on the inside... Super hacky! 

The question is, do we implement this hack, which will cause much more bloat (perhaps around 40% larger size with everything said and done), or do we simply leave it how it is now? 

The features that would be implemented are:

- file-name-formatter
- no-drop-placeholder
- drop-placeholder
- browse-text

So the question is, is it really worth the effort when the default behavior looks just as good?

**Comment by codetheorist** [comment]
Could we not do both, have a "Lite" version without of this added functionality?

**Comment by hoanvh** [comment]
IMO, if BFormFile does not provide more customizations compare to plain HTML `input` tag (+ Bootstrap v5), then I think it's not worth for a dedicate component.
For example, one of a known issue is, if you're a Mac user, then `input` tag cannot display full filename of a file with extension hidden (while it's still there from system perspective such as `ls -la`).
Beside that, i18n has nothing to do here.

**Comment by dwgray** [comment]
Based on the voting from this issue, I re-implemented BFormFile to be closer to what BSV does (#2944), using a hidden <input> control and managing the details with a lot of help from vueUse’s `useFileDialog` and `useDropZone` components.

If you've been waiting for this feature, please take a few minutes to install release >= 0.42.0 and try it out. Please let me know what you think.

And sorry again to those of you who got caught in the initial roll-out of this, I definitely missed some important aspects of the base control the first time around.
